components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/util/ActionMgtAuditLoggerUtil.java

```diff
@@ -0,0 +1,324 @@
+                            authentication.getProperty(Authentication.Property.USERNAME) != null
+                                    ? authentication.getProperty(Authentication.Property.USERNAME).getValue() : ""));
+                    endpointData.put(LogConstants.PASSWORD_FIELD, LoggerUtils.getMaskedContent(
+                            authentication.getProperty(Authentication.Property.PASSWORD) != null
+                                    ? authentication.getProperty(Authentication.Property.PASSWORD).getValue() : ""));
+                    break;
+                case BEARER:
+                    endpointData.put(LogConstants.ACCESS_TOKEN_FIELD, LoggerUtils.getMaskedContent(
+                            authentication.getProperty(Authentication.Property.ACCESS_TOKEN) != null
+                                    ? authentication.getProperty(Authentication.Property.ACCESS_TOKEN).
+                                    getValue() : ""));
+                    break;
+                case API_KEY:
+                    endpointData.put(LogConstants.API_KEY_HEADER_FIELD, LoggerUtils.getMaskedContent(
+                            authentication.getProperty(Authentication.Property.HEADER) != null
+                                    ? authentication.getProperty(Authentication.Property.HEADER).getValue() : ""));
+                    endpointData.put(LogConstants.API_KEY_VALUE_FIELD, LoggerUtils.getMaskedContent(
+                            authentication.getProperty(Authentication.Property.VALUE) != null
+                                    ? authentication.getProperty(Authentication.Property.VALUE).getValue() : ""));
+                    break;
+            }
+        }
+        return endpointData;
+    }
+
+    /**
+     * Operations to be logged.
+     */
+    public enum Operation {
+        ADD("add-action"),
+        UPDATE("update-action"),
+        DELETE("delete-action"),
+        ACTIVATE("activate-action"),
+        DEACTIVATE("deactivate-action");
+
+        private final String logAction;
+
+        Operation(String logAction) {
+
+            this.logAction = logAction;
+        }
+
+        public String getLogAction() {
+
+            return this.logAction;
+        }
+    }
+
+    /**
+     * Action management related log constants.
+     */
+    private static class LogConstants {
+
+        public static final String ACTION_TYPE_FIELD = "ActionType";
+        public static final String ACTION_NAME_FIELD = "ActionName";
+        public static final String ACTION_ID_FIELD = "ActionId";
+        public static final String ACTION_DESCRIPTION_FIELD = "ActionDescription";
+        public static final String ACTION_STATUS_FIELD = "ActionStatus";
+        public static final String ENDPOINT_CONFIG_FIELD = "EndpointConfiguration";
+        public static final String ENDPOINT_URI_FIELD = "EndpointUri";
+        public static final String ENDPOINT_AUTHENTICATION_SCHEME_FIELD = "EndpointAuthenticationScheme";
```

Comment:
Shall we just say AuthenticationScheme ? This is anyway within EndpointConfiguration json object

Reply:
This is updated with b5355995d46dbafa80bb825fa0da60e5bd5062ee